File: tutorial/code/java/Brightness.java
import com.zilogic.zio.*;

class Brightness {
    public static void main(String args[]) 
	throws ProtocolException, InterruptedException {

	Agent agent = new Agent("/dev/ttyUSB0");
	PWM pwm = new PWM(agent); // <1>
	int ledPin = 0x1; // <2>

	pwm.setFreq(ledPin, 2); // <3>
	pwm.setDuty(ledPin, 0); // <4>
	pwm.start(ledPin); // <5>
	
	while (true) {
	    for (int i = 0; i < 100; i += 3) {
		pwm.setDuty(ledPin, i);
		Thread.sleep(10);
	    }
	}
    }
}

File: tutorial/code/python/brightness.py
import zio
import time

agent = zio.Agent("/dev/ttyUSB0")
pwm = zio.PWM(agent) # <1>
pins = [0] # <2>

pwm.set_freq(pins, 2) # <3>
pwm.set_duty(pins, 0) # <4>
pwm.start(pins) # <5>

while (1):
    for i in range(0, 100, 3):
        pwm.set_duty(pins, i)
        time.sleep(0.01)
